src/api/apiClient.ts

```diff
@@ -0,0 +1,141 @@
+import AppAxiosInstance from '../auth/axios';
+import { LeaderboardItem } from '../components/leaderboard/ducks/types';
+
+export interface ApiExtraArgs {
+  readonly apiClient: ApiClient;
+}
+
+export interface ApiClient {
+  readonly getUsersLeaderboard: (
+    previousDays: number,
+  ) => Promise<LeaderboardItem[]>;
+  readonly getTeamsLeaderboard: (
+    previousDays: number,
+  ) => Promise<LeaderboardItem[]>;
+}
+
+enum ApiClientRoutes {
+  USERS_LEADERBOARD = '/api/v1/leaderboard/users',
+  TEAMS_LEADERBOARD = '/api/v1/leaderboard/teams',
+}
+
+const getUsersLeaderboard = (
+  previousDays: number,
+): Promise<LeaderboardItem[]> => {
+  // return AppAxiosInstance.get(ApiClientRoutes.USERS_LEADERBOARD, {
+  //   params: {
+  //     previousDays,
+  //   },
+  // }).then((response) => response.data.users);
+  switch(previousDays) {
```

Comment:
Seems like this switch statement can get thrown into a helper function pretty easily